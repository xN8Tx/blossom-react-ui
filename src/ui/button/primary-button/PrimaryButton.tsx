import React from 'react';

import type { ButtonPropsType } from '../index.types';

import style from './PrimaryButton.module.scss';

export function PrimaryButton({
  children,
  color,
  onClick,
  disabled = false,
  ...props
}: ButtonPropsType) {
  const className = `${style.Button} ${style[color]}`;

  return (
    <button
      className={className}
      onClick={onClick}
      disabled={disabled}
      {...props}
    >
      {children}
    </button>
  );
}
