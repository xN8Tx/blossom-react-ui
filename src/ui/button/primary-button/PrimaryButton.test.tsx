import React from 'react';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';

import { PrimaryButton } from './PrimaryButton';

const mockOnClick = jest.fn();

describe('primary button test', () => {
  afterEach(() => cleanup());

  it('Should render', () => {
    render(
      <PrimaryButton onClick={mockOnClick} color='blue'>
        Blossom Button
      </PrimaryButton>,
    );

    const button = screen.getByText('Blossom Button');

    expect(button).toBeInTheDocument();
  });
  it('Should click', () => {
    render(
      <PrimaryButton onClick={mockOnClick} color='blue'>
        Blossom Button
      </PrimaryButton>,
    );

    const button = screen.getByText('Blossom Button');
    expect(button).toBeInTheDocument();

    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalled();
  });
  it('Should not click', () => {
    render(
      <PrimaryButton onClick={mockOnClick} color='blue' disabled={true}>
        Blossom Button
      </PrimaryButton>,
    );

    const button = screen.getByText('Blossom Button');
    expect(button).toBeInTheDocument();

    fireEvent.click(button);

    expect(mockOnClick).not.toHaveBeenCalled();
  });
  it('Snapshot test', () => {
    render(
      <PrimaryButton onClick={mockOnClick} color='blue' disabled={true}>
        Blossom Button
      </PrimaryButton>,
    );

    const button = screen.getByText('Blossom Button');
    expect(button).toBeInTheDocument();

    expect(button).toMatchSnapshot();
  });
});
