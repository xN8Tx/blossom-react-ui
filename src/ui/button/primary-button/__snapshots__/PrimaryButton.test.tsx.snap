// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`primary button test Snapshot test 1`] = `
<button
  class="Button blue"
  disabled=""
>
  Blossom Button
</button>
`;
